TinyMCE.Blazor/Editor.razor

```diff
@@ -18,186 +20,307 @@
-    string acc = "";
-    bool remove = streamMap.TryGetValue(streamId, out acc);
-    acc = acc + content;
-    if (index == size) {
-      if (remove) { streamMap.Remove(streamId); }
-      callback(acc);
+        get => _disable;
+        set
+        {
+            _setDisable = true;
+            if (_disable != value)
+            {
+                _disable = value;
+                _ = ChangeMode(value);
+            }
+        }
     }
-    else
+
+    [JSInvokable("GetValue")]
+    public Task<string> ReturnValue()
     {
-      streamMap[streamId] = acc;
-    }
-  }
-
-  protected async Task PushValue(string nextVal)
-  {
-    int chunks = (nextVal.Length / _chunkSize) + 1;
-    string streamId = String.Format("{0:d9}", (DateTime.Now.Ticks / 10) % 1000000000);
-    for (int i = 0; i < chunks; i++) {
-      string chunk = nextVal.Substring( _chunkSize * i, nextVal.Length < (_chunkSize * ( i + 1)) ? (nextVal.Length - (_chunkSize * i)) : _chunkSize);
-      await JSRuntime.InvokeVoidAsync("tinymceBlazorWrapper.updateValue", Id, streamId, chunk, i + 1, chunks);
-    }
-  }
-
-  protected async Task ChangeMode(bool disabled)
-  {
-    await JSRuntime.InvokeVoidAsync("tinymceBlazorWrapper.updateMode", Id, disabled);
-  }
-    
-  protected override async Task OnAfterRenderAsync(bool firstRender)
-  {
-    if (firstRender)
-    {
-      blazorConf = new Dictionary<string, object>();
-      blazorConf.Add("inline", Inline);
-      string tinySrc = ScriptSrc != null ? ScriptSrc : $"https://cdn.tiny.cloud/1/{ApiKey}/tinymce/{CloudChannel}/tinymce.min.js";
-      blazorConf.Add("src", tinySrc);
-      blazorConf.Add("jsConf", JsConfSrc);
-      blazorConf.Add("modelEvents", ModelEvents);
-      Dictionary<string, object> merged = Merge(Conf, AdditionalAttributes);
-      blazorConf.Add("conf", merged);
-      if (!_setDisable && merged.ContainsKey("readonly"))
-      {
-        _disable = merged["readonly"] is bool ? (bool)merged["readonly"] : false;
-      }
-      blazorConf.Add("disabled", Disable);
-      // TBD: Load the events
-        @*foreach (KeyValuePair<string, object> kvp in AdditionalAttributes)
+        return Task.FromResult(Value);
+    }
+
+    [JSInvokable("UpdateModel")]
+    public void OnChangeContent(string streamId, int index, string val, int size)
+    {
+        PushUpdate(streamId, index, val, size, (content) =>
+        {
+            _value = content;
+            ValueChanged.InvokeAsync(content);
+        });
+    }
+
+    [JSInvokable("UpdateText")]
+    public void OnChangeText(string streamId, int index, string val, int size)
+    {
+        PushUpdate(streamId, index, val, size, (content) =>
+        {
+            _text = content;
+            TextChanged.InvokeAsync(content);
+        });
+    }
+
+    /*
+     * By default, hooked up with TinyMce's onchange event;
+     */
+
+    [JSInvokable("OnChange")]
+    public void OnChange()
+    {
```

Comment:
This is called straight from the editor but the new value is not available yet. This could lead to weird cases were we notify changes have happened but we are still holding the old value. Same thing happens with `OnInput`

Reply:
These two methods are called from TinyMCE's `change` and `input` events. But all they do is notify `EditContext` for validation and prepend classes to `ClassName`. It doesn't affect values.

Reply:
The validation would be the problem if the `EditContext` is notified of the change and tries validating against the old value instead of the new one. I think it would make more sense to notify until after you have the latest value. I think this should happen in `OnChangeContent` and in the `Value` setter.

Reply:
You are right, it validates against old value!